clients/da-vinci-client/src/main/java/com/linkedin/davinci/consumer/VeniceChangelogConsumerImpl.java

```diff
@@ -302,6 +303,36 @@ public CompletableFuture<Void> subscribeAll() {
     return this.subscribe(allPartitions);
   }
 
+  @Override
+  public CompletableFuture<Void> seekToTimestamp(Map<Integer, Long> timeStamps) {
+    // Get the latest change capture topic
+    storeRepository.refresh();
+    Store store = storeRepository.getStore(storeName);
+    int currentVersion = store.getCurrentVersion();
+    String topic = Version.composeKafkaTopic(storeName, currentVersion) + ChangeCaptureView.CHANGE_CAPTURE_TOPIC_SUFFIX;
```

Comment:
One thing needs to be clarified here is: is it possible this timestamp user will derive from the time when user is doing RT producing. As we query offset based on this timestamp for kafka topic data generate by venice, so there might be time gap here.

Reply:
It's a good question.  The problem is that it's not going to be very easy to seek by user timestamp.  User timestamp need not line up with wall clock time or ordering.  So in order for this to work reliably, we're gonna have to scan the entirety of the topic to find a record with a matching timestamp. Moreover, events can share a timestamp, so it's kind of vague where you'd like to land.

Add to it, we can't even binary search topics for the right record for the above reason and the inability to do math on offsets in Xinfra.

We can document it in the javadoc, but I don't see this working easily.  I think wallclock is best we can do given the above limitations.  Maybe later we can add a 'findVeniceLogicalTimestamp' kind of api?  Not sure.

Reply:
Yeah, we can clarify on the java doc. As there is always time gap between the topic produced by venice and topic produced by users. Searching through kafka records by our own is bit complicated, so I think we do not need to support this at this moment.

Reply:
Jotting this down for future work after thinking on this more, some points: 

- With the current seek API's a user could duplicate the logic we'd just put into the client with the current API's (filtering until you find what you need via seekToEndOfPush).  

- We don't expose produce metadata right now in the pubsubMessage since it's contained the KafkaMessageEnvelope that we strip away.  So for a future api where we'd seek to some producerTimestamp we'll probably need to backfill that piece of information to the PubSubMessage along with the new interface api's.

- Doing this for a batch of partitions might be a bit confusing if a set of producers aren't using wall clock time, but some other input from an upstream source, maybe it should be discouraged?

- Assuming RT write rate is small, and the hybrid rewind isn't enormous, it might not be that much data that need be consumed client side to find the right offset.  Maybe for most cases it would actually be reasonable?  Not sure.

Reply:
Nevermind we do provide the timestamp in the pubsubmessage, and looks like it's the producer timestamp, so from the consumer side with the current api's I think this would work nicely.

We'd later need to add a getter for upstream producer timestamp.

Reply:
Updated javadoc to reflect this.